### PR TITLE
Update Pubsub Documentation

### PR DESCRIPTION
**Pull request by blowmage**
Overhaul the documentation with parameter definitions and several code examples.

### CODE REVIEW

File: lib/gcloud/pubsub.rb [hunk 1]
```diff
@@ -43,14 +56,168 @@ def self.pubsub project = nil, keyfile = nil
...
+  #   pubsub = Gcloud.pubsub
+  #
+  #   topic = pubsub.topic "my-topic"
+  #   msg = topic.publish "new-message"
+  #
+  # Messages can also be published with attributes:
+  #
+  #   require "glcoud/pubsub"
+  #
+  #   pubsub = Gcloud.pubsub
```
Comment (jgeewax) [inline lib/gcloud/pubsub.rb]: Can we condense these a bit? Feels like they're extra spaced apart -- almost needlessly so.

Comment (blowmage) [inline lib/gcloud/pubsub.rb]: You mean the empty line between `pubsub =` and `topic =`?

Comment (jgeewax) [inline lib/gcloud/pubsub.rb]: Yessir.

Particular for the 3-line examples, where each line gets a newline.

Comment (blowmage) [inline lib/gcloud/pubsub.rb]: K, will update.

**Pull request merged**